x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentNodeService.java

```diff
@@ -432,15 +435,48 @@ private static StartTrainedModelDeploymentAction.TaskParams createStartTrainedMo
         );
     }
 
-    private boolean isAssignmentOnShuttingDownNode(
-        RoutingInfo routingInfo,
-        String deploymentId,
+    private boolean shouldGracefullyShutdownDeployment(
+        TrainedModelAssignment trainedModelAssignment,
         Set<String> shuttingDownNodes,
         String currentNode
     ) {
-        return deploymentIdToTask.containsKey(deploymentId)
-            && routingInfo.getState() == RoutingState.STOPPING
-            && shuttingDownNodes.contains(currentNode);
+        RoutingInfo routingInfo = trainedModelAssignment.getNodeRoutingTable().get(currentNode);
+
+        if (routingInfo == null) {
+            return false;
+        }
```

Comment:
With the current calling point for `shouldGracefullyShutdownDeployment` it's impossible for this test to pass and return `false` because there a check for `routingInfo != null` above where it's called.

So I guess this check is just for future-proofing. In that case I think it's better that it returns `true`, because if a node isn't mentioned in a model's routing table then in general it shouldn't be running there.

Reply:
Yeah good point. In the changes here I decided to start passing in `TrainedModelAssignment` since I needed it for a few other things and figured I should do a null check. I'll switch it to `true` 👍